Assets/MRTK/SDK/Editor/Inspectors/UX/InteractiveElement/TrackedStatesInspector.cs

```diff
@@ -0,0 +1,282 @@
+            if (stateEventConfiguration.objectReferenceValue == null && StateHasExistingEventConfiguration(stateName.stringValue))
+            {
+                // Draw a button in the state container titled 'Add State Event'. On button press, an instance of the matching
+                // event configuration scriptable is created and assigned to the state
+                if (GUILayout.Button("Add " + stateName.stringValue + " State Events"))
+                {
+                    CreateEventScriptable(stateEventConfiguration, stateName.stringValue);
+                }
+            }
+            else if (stateEventConfiguration.objectReferenceValue != null)
+            {
+                string stateFoldoutID = stateName.stringValue + "EventConfiguration";
+
+                // Draw a foldout for the event configuration
+                if (InspectorUIUtility.DrawSectionFoldoutWithKey(stateName.stringValue + " State Events", stateFoldoutID, MixedRealityStylesUtility.TitleFoldoutStyle, false))
+                {
+                    // Draw the events for the associated state if this state has an event configuration
+                    DrawStateEventsScriptableSubEditor(stateEventConfiguration);
+                }
+            }
+        }
+
+        // Check if the given state has a valid event configuration to render
+        private bool StateHasExistingEventConfiguration(string stateName)
+        {
+            // Get the list of the subclasses for BaseInteractionEventConfiguration and find the
+            // event configuration that contains the given state name
+            var eventConfigurationTypes = TypeCacheUtility.GetSubClasses<BaseInteractionEventConfiguration>();
+            var eventConfigType = eventConfigurationTypes.Find(t => t.Name.StartsWith(stateName));
+
+            // If there is associated event configuration for the state, then it is valid
+            if (eventConfigType != null)
+            {
+                return true;
+            }
+            // If an associated class for the state could not be found then the event config is not valid
+            else
+            {
+                return false;
+            }
+        }
+
+        // Create an instance of an associated event scriptable object given the state.
+        private void CreateEventScriptable(SerializedProperty eventConfiguration, string stateName)
+        {
+            string className = stateName + "InteractionEventConfiguration";
+
+            // If the event configuration for the state is currently, then create a scrpitable instance
+            if (eventConfiguration.objectReferenceValue == null)
+            {
+                // Initialize the associated scriptable object event configuration with the correct state 
+                eventConfiguration.objectReferenceValue = ScriptableObject.CreateInstance(className);
+                eventConfiguration.objectReferenceValue.name = stateName + "EventConfiguration";
```

Comment:
Why is it necessary to append "EventConfiguration" here? Are you using string values to somehow encode information about the underlying type?

Reply:
It is not necessary to append "EventConfiguration".  The purpose of this line:

`eventConfiguration.objectReferenceValue.name = stateName + "EventConfiguration";`

is to just add a label the newly created scriptable object instance.